### PR TITLE
1.20.5/6

### PR DESCRIPTION
**Pull request by wagyourtail**

### CODE REVIEW

File: src/api/java/baritone/api/utils/BlockOptionalMeta.java [hunk 1]
```diff
@@ -218,43 +216,21 @@ private static VanillaPackResources getVanillaServerPack() {
...
-                                            .create(LootContextParamSets.BLOCK)
-                            ).withOptionalRandomSeed(1L)
-                                    .create(null),
-                            stack -> items.add(stack.getItem())
-                    );
+                    LootParams.Builder lv5 = new LootParams.Builder(lv2)
+                        .withParameter(LootContextParams.ORIGIN, Vec3.ZERO)
+                        .withParameter(LootContextParams.BLOCK_STATE, b.defaultBlockState())
+                        .withParameter(LootContextParams.TOOL, new ItemStack(Items.NETHERITE_PICKAXE, 1));
+                    getDrops(block, lv5).stream().map(ItemStack::getItem).forEach(items::add);
```
Comment (wagyourtail) [inline src/api/java/baritone/api/utils/BlockOptionalMeta.java:244]: this didn't work. registry is missing LootTable entries, so needs to be re-built somehow.

Comment (rfresh2) [reply src/api/java/baritone/api/utils/BlockOptionalMeta.java:244]: i put up a gist for how i've initialized these registries for client access: https://gist.github.com/rfresh2/697c3996dbfdfe595c3322f2adfaa4a6 might be possible to simplify this further

Comment (HackThePyramids) [reply src/api/java/baritone/api/utils/BlockOptionalMeta.java:244]: does it work now?

Comment (wagyourtail) [reply src/api/java/baritone/api/utils/BlockOptionalMeta.java:244]: that appears to work. good enough for now. thanks!
just had to change one line to not deadlock with where I run it :)

**Comment by rfresh2** [comment]
Found a few issues so far:

* Forge:
	* Mixins are being remapped, they need to be kept in mojmap like neoforge now
* NeoForge:
	* `processResources` gradle task needs to be updated to point to neoforge.mods.toml
	* [This Neoforge patch](https://github.com/neoforged/NeoForge/blob/8f8559b0fcdc4ec4577dee9186d9d9f72de890bf/patches/net/minecraft/client/player/LocalPlayer.java.patch#L75-L76) is causing an error in `MixinClientPlayerEntity$isAllowFlying` as they replaced the injection point. Might become more difficult to maintain mod loader agnostic mixins if neo continues making more src patches like this

**Comment by ZacSharp** [comment]
As mentioned by @Edouard127 in #4354 we also have to update our dependency declarations for neo.
EDIT: Actually, they only mentioned the rename and I picked up the dependency format change from the link they provided.

**Comment by Sayrix** [comment]
Seems to work pretty good on Fabric 1.20.6

**Review by Sayrix — APPROVED** [review]

File: neoforge/src/main/resources/META-INF/neoforge.mods.toml [hunk 2]
```diff
@@ -35,6 +35,6 @@ A Minecraft pathfinder bot.
 modId="minecraft"
 mandatory=true
```
Comment (ZacSharp) [inline neoforge/src/main/resources/META-INF/neoforge.mods.toml]:
```suggestion
type="required"
```
Warning: untested

**Comment by SSolarite** [comment]
holy SHIT is that sayrix from nerd discord

**Comment by Sayrix** [comment]
> holy SHIT is that sayrix from nerd discord

Yes.

https://github.com/cabaletta/baritone/assets/43046854/64d6a290-0178-4998-984f-647e90d85f93

**Comment by enygmate** [comment]
> > holy SHIT is that sayrix from nerd discord
> 
> Yes.
> 
>  Sparkles.in.japan.mp4

HOLY SHIT IS THAT SAYRIX FROM U'World MC ???

**Comment by unsaltedbutter0** [comment]
Hi, I'm having an issue with selection or rather baritone inverting it. I'm using baritone on fabric and this happens without any more mods present
Screenshot 2024-07-27 at 14 08 13
In the picture, u can see what I mean by "inverting".

Edit:
I don't know why I didn't try this sooner but after resetting the settings it somehow started working. I've tried to find the previous settings but with no luck. I might have them in my backup but I do not have any way of checking it right now.

**Pull request merged**